### PR TITLE
Update CI workflow to include publishing to PyPI based on a GitHub release

### PR DESCRIPTION
**Pull request by stevejalim**
Add GH Action workflow to auto-upload to PyPI when a release is made.

DONE ~Note that this will not work until a PyPI token has been set as an environment secret for the Action, with the name `PYPI_API_TOKEN`~

### CODE REVIEW

**Comment by stevejalim** [comment]
@pmac @robhudson This is ready for merge - the API key is in place

**Comment by robhudson** [comment]
r+wc

**Review by pmac — APPROVED** [review]
⭐

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -49,3 +55,27 @@ jobs:
...
+    needs:
+      - test
+      - lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - name: Install lightest dependencies for successful packaging
```
[the remark this answers is not in the document]
Comment (stevejalim) [reply .github/workflows/ci.yml:71]: No, lightest as in minimal deps, but i appreciate the laser eyes

**Pull request merged**